packages/dialect-react-ui/hooks/useDappAudience.ts

```diff
@@ -28,22 +30,31 @@ interface UseDappAudienceValue {
 
 export default function useDappAudience({
   notificationTypeId,
+  addressTypes,
   refreshInterval = DEFAULT_ADDRESSES_REFRESH_INTERVAL,
 }: UseDappAudienceParams): UseDappAudienceValue {
   const {
     subscriptions: notificationsSubscriptions,
     isFetching: isFetchingSubscriptions,
     errorFetching: errorFetchingNotificationSubscriptions = null,
   } = useDappNotificationSubscriptions();
+  let { addresses } = useDappAddresses({
+    refreshInterval,
+  });
   const {
-    addresses,
     isFetching: isFetchingAddresses,
     errorFetching: errorFetchingAddresses = null,
   } = useDappAddresses({
     refreshInterval,
   });
   const isFetching = isFetchingSubscriptions || isFetchingAddresses;
 
+  if (addressTypes) {
+    addresses = addresses.filter((addr) =>
+      addressTypes.includes(addr.address.type)
+    );
+  }
```

Comment:
looks soo mutable, can we change it to immutable expression? Afraid that someone might get struggled while debugging something here

Reply:
hey @zhelezkov, changed this part a bit, wdyt?